src/core/lib/gpr/cpu_posix.cc

```diff
@@ -71,7 +70,7 @@ unsigned gpr_cpu_current_cpu(void) {
   unsigned int* thread_id =
       static_cast<unsigned int*>(pthread_getspecific(thread_id_key));
   if (thread_id == nullptr) {
-    thread_id = static_cast<unsigned int*>(gpr_malloc(sizeof(unsigned int)));
+    thread_id = static_cast<unsigned int*>(malloc(sizeof(unsigned int)));
```

Comment:
add comments on why we are making the switch

Reply:
Done.